mvcc/kvstore.go

```diff
@@ -293,9 +293,15 @@ func (s *store) Compact(rev int64) (<-chan struct{}, error) {
 	return ch, nil
 }
 
-func (s *store) Hash() (uint32, error) {
+func (s *store) Hash() (uint32, int64, error) {
 	s.b.ForceCommit()
-	return s.b.Hash()
```

Comment:
we need to hold the store lock through this method. or we can still get bad revision. we need to ensure there are no  modification happens during getting hash and getting the revision.